CONTRIBUTING.md

```diff
@@ -0,0 +1,173 @@
+* [Saying why it failed in simple turns](https://github.com/Regression-Games/RegressionBot/blob/e2bddcc37950a9f2ee5718ff5aa4dda549c5f008/lib/RGBot.js#L1102)
+
+**Measurement Approach:** Internal Audit
+
+**Most recent test results:** 
+
+(Dec 27, 2022) Out of **17** explicit errors, **9** had a next steps or more info provided = **53%** success rate.
+
+Failed cases:
+* [When attacking an enemy fails](https://github.com/Regression-Games/RegressionBot/blob/e2bddcc37950a9f2ee5718ff5aa4dda549c5f008/lib/RGBot.js#L735)
+* [When approaching a block fails](https://github.com/Regression-Games/RegressionBot/blob/e2bddcc37950a9f2ee5718ff5aa4dda549c5f008/lib/RGBot.js#L956)
+* [When equipping a tool fails](https://github.com/Regression-Games/RegressionBot/blob/e2bddcc37950a9f2ee5718ff5aa4dda549c5f008/lib/RGBot.js#L1005)
+* [When equipping a tool fails](https://github.com/Regression-Games/RegressionBot/blob/e2bddcc37950a9f2ee5718ff5aa4dda549c5f008/lib/RGBot.js#L1099)
+* [When approaching and digging a block fails](https://github.com/Regression-Games/RegressionBot/blob/e2bddcc37950a9f2ee5718ff5aa4dda549c5f008/lib/RGBot.js#L1172)
+* [When collecting an item fails](https://github.com/Regression-Games/RegressionBot/blob/e2bddcc37950a9f2ee5718ff5aa4dda549c5f008/lib/RGBot.js#L1332)
+* [When dropping an item fails](https://github.com/Regression-Games/RegressionBot/blob/e2bddcc37950a9f2ee5718ff5aa4dda549c5f008/lib/RGBot.js#L1417)
+* [Error when crafting an item](https://github.com/Regression-Games/RegressionBot/blob/e2bddcc37950a9f2ee5718ff5aa4dda549c5f008/lib/RGBot.js#L1498)
+
+## Link to Survey for Fact Checking
+
+https://d7y6yysps34.typeform.com/to/feG9z7kp
+
+## Design Patterns
+
+### Awaits, Promises, and Callbacks
+
+Most of our functions and design patterns should use Promises + `await` when possible. For example:
+
+**DON'T IMPLEMENT**
+```javascript
+bot.findAndDigBlock('dirt', (block) => {
+    ...
+})
+```
+**DON'T PROVIDE AS AN EXAMPLE**
+```javascript
+bot.findAndDigBlock('dirt')
+  .then((block) => {...})
+```
+
+**DO**
+```javascript
+let block = await bot.findAndDigBlock('dirt')
+```
+
+In the case where a user-provided block of code **needs** to be passed to a library function, this is fine. This should really only happen when the operation requires some sort of context that needs managing (such as opening a chest, e.g. there is a before and after operation). For example:
+
+**DO**
+```javascript
+bot.openAndInteractWithChest(chest, (chest) => {
+    chest.place(...)
+});
+```
+
+In the cases of these context-aware operations (opening and closing a chest, mounting and unmounting a horse, etc..) we should still provide the context-unaware functions, that the user can use themselves). These are useful for sending new players quick code examples that they can understand.
+
+**ALSO DO**
+```javascript
+let openChest = bot.openChest(chest);
```

Comment:
we should should then how to use try/finally for any cases like this where the follow up API call is required to properly manage state

Reply:
Added!